### ISSUE TITLE
Upgrade suggestion

### ISSUE THREAD

**Issue by Melethrion** [kacpak/Auto-Sell-Grey-and-Repair-Equipement#17]
Hey there,

with the new function I'd suggest an update from bagslot-iteration to _C_MerchantFrame.SellAllJunkItems()_ - which would surpass the limited count on items able to be sold at once when using the iteration method and thus would sell all greys at once.

Cheers

**Comment by Strik3ria** [comment]
Will have to rework how we get the total value of sold items as the new function does not return this information. Will have to use gold in bag before selling - gold in bag after selling I think?

**Comment by Melethrion** [comment]
nah, i tried it, you can keep everything except how you sell tiems. just let it still iterate through the junk to count the value then sell via command after that loop, like:

	-- Auto Sell Grey Items
	totalPrice = 0	
	for myBags = 0,4 do
		for bagSlots = 1, C_Container.GetContainerNumSlots(myBags) do
			CurrentItemLink = C_Container.GetContainerItemLink(myBags, bagSlots)
			if CurrentItemLink then
				_, _, itemRarity, _, _, _, _, _, _, _, itemSellPrice = GetItemInfo(CurrentItemLink)
				itemInfo = C_Container.GetContainerItemInfo(myBags, bagSlots)
				if itemRarity == 0 and itemSellPrice ~= 0 then
					totalPrice = totalPrice + (itemSellPrice * itemInfo.stackCount)
					--C_Container.UseContainerItem(myBags, bagSlots)
					--PickupMerchantItem()
				end
			end
		end
	end
	if totalPrice ~= 0 then
		C_MerchantFrame.SellAllJunkItems() -- here
		DEFAULT_CHAT_FRAME:AddMessage("Items were sold for "..GetCoinTextureString(totalPrice), 255, 255, 255)
	end

_etc._

**Comment by Melethrion** [comment]
thinking about it, of course you could as well just minimize the code to the new command and dump the loop by just comparing the two mentioned values of gold in bags. i was actually being lazy. :D but this as a first step would at least finally get rid of that hidden cap of 12 or so items that do sell at the most if bags are really cluttered...

p.s.: great. now i've got to resist the urge to just do that as well. it's itching. a lot. >.<

**Comment by rcoones** [comment]
Sounds like you should just make a PR =P

**Comment by rcoones** [comment]
I think I am going to try some things with this tomorrow =D see what I can come up with

**Comment by rcoones** [comment]
Did some more testing today, found that we can use throttling to sell all items in a loop still. Using the sellalljunkitems function really makes getting the amount you sold greys for very difficult timing wise. The playey money event only fires so often and it favors batching money events and therefore our repair and sell junk processes both happen in the same player money event. I tried using some delaying but it still mixed the two actions values together. I ended up using a more recursive function with a C_Timer and it works much better. When I have some time I will update my PR.
